src/vm/builtins/builtin_runner/range_check.zig

```diff
@@ -296,3 +329,166 @@ test "used instances" {
         try builtin.getUsedInstances(memory_segment_manager),
     );
 }
+
+test "Range Check: get usage for range check" {
+
+    // given
+    var builtin = RangeCheckBuiltinRunner.new(8, 8, true);
+    const allocator = std.testing.allocator;
+    var seg = try MemorySegmentManager.init(allocator);
+    defer seg.deinit();
+
+    try memoryFile.setUpMemory(seg.memory, std.testing.allocator, .{
+        .{ .{ 0, 0 }, .{1} },
+        .{ .{ 0, 1 }, .{2} },
+        .{ .{ 0, 2 }, .{3} },
+        .{ .{ 0, 3 }, .{4} },
+    });
+    defer seg.memory.deinitData(std.testing.allocator);
+    const res = builtin.getRangeCheckUsage(seg.memory);
+    const four: usize = 4;
+    const zero: usize = 0;
+    // assert
+    try std.testing.expectEqual(
+        res.?[1],
+        four,
+    );
+    try std.testing.expectEqual(
+        res.?[0],
+        zero,
+    );
```

Comment:
Here I know this is confusing in Zig but the expected value should come first. Also, if you use a tuple for the result of the function you could have a stronger test by doing:

```zig
try std.testing.expectEqual(
    @as(std.meta.Tuple(&.{ usize, usize }), .{ 0, 4 }),
    res.?
);
```
so as to test the complete result of the function in a single expect and thus be sure of the entirety of what we are testing.

If you ultimately choose to stay with an array, it is probably preferable to test the slice with the `expectEqualSlices` method in order to also be sure of the entire slice that you have as output.